thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java

```diff
@@ -0,0 +1,228 @@
+
+  public Map<String, Integer> getFunctionFilterMap() {
+    return functionFilterMap;
+  }
+
+  public void setFunctionFilterMap(Map<String, Integer> functionFilterMap) {
+    this.functionFilterMap = functionFilterMap;
+  }
+
+  public Map<String, Integer> getDatasetFilterMap() {
+    return datasetFilterMap;
+  }
+
+  public void setDatasetFilterMap(Map<String, Integer> datasetFilterMap) {
+    this.datasetFilterMap = datasetFilterMap;
+  }
+
+  public Map<String, Integer> getMetricFilterMap() {
+    return metricFilterMap;
+  }
+
+  public void setMetricFilterMap(Map<String, Integer> metricFilterMap) {
+    this.metricFilterMap = metricFilterMap;
+  }
+
+  public Map<String, Map<String, Integer>> getDimensionFilterMap() {
+    return dimensionFilterMap;
+  }
+
+  public void setDimensionFilterMap(Map<String, Map<String, Integer>> dimensionFilterMap) {
+    this.dimensionFilterMap = dimensionFilterMap;
+  }
+
+  public static List<MergedAnomalyResultDTO> applySearchFilters(List<MergedAnomalyResultDTO> anomalies, SearchFilters searchFilters) {
+    if (searchFilters == null) {
+      return anomalies;
+    }
+    Map<String, Integer> statusFilterMap = searchFilters.getStatusFilterMap();
+    Map<String, Integer> functionFilterMap = searchFilters.getFunctionFilterMap();
+    Map<String, Integer> datasetFilterMap = searchFilters.getDatasetFilterMap();
+    Map<String, Integer> metricFilterMap = searchFilters.getMetricFilterMap();
+    Map<String, Map<String, Integer>> dimensionFilterMap = searchFilters.getDimensionFilterMap();
+    List<MergedAnomalyResultDTO> filteredAnomalies = new ArrayList<>();
+    for (MergedAnomalyResultDTO mergedAnomalyResultDTO : anomalies) {
+      boolean passed = true;
+      // check status filter
+      AnomalyFeedback feedback = mergedAnomalyResultDTO.getFeedback();
+      if (feedback != null) {
+        String status = feedback.getStatus().toString();
+        passed = passed && checkFilter(statusFilterMap, status);
+      }
+      // check status filter
+      String functionName = mergedAnomalyResultDTO.getFunction().getFunctionName();
+      passed = passed && checkFilter(functionFilterMap, functionName);
+      // check datasetFilterMap
+      String dataset = mergedAnomalyResultDTO.getCollection();
+      passed = passed && checkFilter(datasetFilterMap, dataset);
+      // check metricFilterMap
+      String metric = mergedAnomalyResultDTO.getMetric();
+      passed = passed && checkFilter(metricFilterMap, metric);
+      // check dimensionFilterMap
+      if (dimensionFilterMap != null) {
+        DimensionMap dimensions = mergedAnomalyResultDTO.getDimensions();
+        for (String dimensionName : dimensions.keySet()) {
+          if (dimensionFilterMap.containsKey(dimensionName)) {
+            String dimensionValue = dimensions.get(dimensionName);
+            passed = passed && checkFilter(dimensionFilterMap.get(dimensionName), dimensionValue);
+          }
+        }
+      }
+      if (passed) {
```

Comment:
Should we make the check fails fast? When an anomaly does not pass a filter, we skip to the next anomaly immediately.